### ISSUE TITLE
Error after upgrade from 5.1.2 to 5.2.0 when opening modal for edit data

### ISSUE THREAD

**Issue by mrjohnr** [valor-software/ng2-bootstrap#5399]
ngx-bootstrap: 5.2.0
Angular: 8.2.7
Bootstrap: 4.3.1

After I've upgraded my project from ngx 5.1.2 to 5.2.0 when i try to open a edit data modal  I found about 15000 lines of errors + browser freezes,errors are like below
tried [this](https://github.com/valor-software/ngx-bootstrap/issues/3809) but no fix
switching back to 5.1.2 there is no error

`i18n.js:26 [Deprecation] Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check https://xhr.spec.whatwg.org/.
s3forecast.i18n.init @ i18n.js:26
core.js:25637 Angular is running in the development mode. Call enableProdMode() to enable the production mode.
client:52 [WDS] Live Reloading enabled.
EditRezComponent.html:77 ERROR Error: ExpressionChangedAfterItHasBeenCheckedError: Expression has changed after it was checked. Previous value: 'model: 1'. Current value: 'model: 2'.
    at viewDebugError (core.js:19628)
    at expressionChangedAfterItHasBeenCheckedError (core.js:19616)
    at checkBindingNoChanges (core.js:19832)
    at checkNoChangesNodeInline (core.js:29536)
    at checkNoChangesNode (core.js:29521)
    at debugCheckNoChangesNode (core.js:30125)
    at debugCheckDirectivesFn (core.js:30053)
    at Object.eval [as updateDirectives] (EditRezComponent.html:77)
    at Object.debugUpdateDirectives [as updateDirectives] (core.js:30042)
    at checkNoChangesView (core.js:29420)
View_EditRezComponent_4 @ EditRezComponent.html:77
push../node_modules/@angular/core/fesm5/core.js.DebugContext_.logError @ core.js:30270
push../node_modules/@angular/core/fesm5/core.js.ErrorHandler.handleError @ core.js:4050
(anonymous) @ core.js:27136
ZoneDelegate.invoke @ zone.js:396
Zone.run @ zone.js:153
push../node_modules/@angular/core/fesm5/core.js.NgZone.runOutsideAngular @ core.js:26214
push../node_modules/@angular/core/fesm5/core.js.ApplicationRef.tick @ core.js:27136
(anonymous) @ core.js:26997
ZoneDelegate.invoke @ zone.js:396
onInvoke @ core.js:26255
ZoneDelegate.invoke @ zone.js:395
Zone.run @ zone.js:153
push../node_modules/@angular/core/fesm5/core.js.NgZone.run @ core.js:26169
next @ core.js:26997
schedulerFn @ core.js:23734
push../node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.__tryOrUnsub @ Subscriber.js:192
push../node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.next @ Subscriber.js:130
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber._next @ Subscriber.js:76
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:53
push../node_modules/rxjs/_esm5/internal/Subject.js.Subject.next @ Subject.js:47
push../node_modules/@angular/core/fesm5/core.js.EventEmitter.emit @ core.js:23718
checkStable @ core.js:26224
onHasTask @ core.js:26268
ZoneDelegate.hasTask @ zone.js:451
ZoneDelegate._updateTaskCount @ zone.js:472
Zone._updateTaskCount @ zone.js:294
Zone.runTask @ zone.js:215
drainMicroTaskQueue @ zone.js:611
ZoneTask.invokeTask @ zone.js:517
invokeTask @ zone.js:1671
globalZoneAwareCallback @ zone.js:1697
Show 2 more frames
EditRezComponent.html:77 ERROR CONTEXT DebugContext_ {view: {…}, nodeIndex: 8, nodeDef: {…}, elDef: {…}, elView: {…}}
View_EditRezComponent_4 @ EditRezComponent.html:77
push../node_modules/@angular/core/fesm5/core.js.DebugContext_.logError @ core.js:30270
push../node_modules/@angular/core/fesm5/core.js.ErrorHandler.handleError @ core.js:4055
(anonymous) @ core.js:27136
ZoneDelegate.invoke @ zone.js:396
Zone.run @ zone.js:153
push../node_modules/@angular/core/fesm5/core.js.NgZone.runOutsideAngular @ core.js:26214
push../node_modules/@angular/core/fesm5/core.js.ApplicationRef.tick @ core.js:27136
(anonymous) @ core.js:26997
ZoneDelegate.invoke @ zone.js:396
onInvoke @ core.js:26255
ZoneDelegate.invoke @ zone.js:395
Zone.run @ zone.js:153
push../node_modules/@angular/core/fesm5/core.js.NgZone.run @ core.js:26169
next @ core.js:26997
schedulerFn @ core.js:23734
push../node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.__tryOrUnsub @ Subscriber.js:192
push../node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.next @ Subscriber.js:130
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber._next @ Subscriber.js:76
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:53
push../node_modules/rxjs/_esm5/internal/Subject.js.Subject.next @ Subject.js:47
push../node_modules/@angular/core/fesm5/core.js.EventEmitter.emit @ core.js:23718
checkStable @ core.js:26224
onHasTask @ core.js:26268
ZoneDelegate.hasTask @ zone.js:451
ZoneDelegate._updateTaskCount @ zone.js:472
Zone._updateTaskCount @ zone.js:294
Zone.runTask @ zone.js:215
drainMicroTaskQueue @ zone.js:611
ZoneTask.invokeTask @ zone.js:517
invokeTask @ zone.js:1671
globalZoneAwareCallback @ zone.js:1697`
................
thanks

**Comment by mrjohnr** [comment]
same issue in 5.3.0,reverting to 5.1.2 all works fine
in a code like this
```
...
private modalService: BsModalService
... 
this.modalRef = this.modalService.show(EditRezComponent
..
alert('1');
 this.modalRef.content.outResult.subscribe(res => {
        if (res) {
            alert('2');
...

```
alert '1' is shown but not '2'
**Expression has changed after it was checked. Previous value: 'model: 1'. Current value: 'model: 2'.**
```
EditRezComponent.html:85 ERROR Error:
 ExpressionChangedAfterItHasBeenCheckedError: **Expression has changed after it was checked. Previous value: 'model: 1'. Current value: 'model: 2'.**
    at viewDebugError (core.js:19628)
    at expressionChangedAfterItHasBeenCheckedError (core.js:19616)
    at checkBindingNoChanges (core.js:19832)
    at checkNoChangesNodeInline (core.js:29536)
    at checkNoChangesNode (core.js:29521)
    at debugCheckNoChangesNode (core.js:30125)
    at debugCheckDirectivesFn (core.js:30053)
    at Object.eval [as updateDirectives] (EditRezComponent.html:85)
    at Object.debugUpdateDirectives [as updateDirectives] (core.js:30042)
    at checkNoChangesView (core.js:29420)
View_EditRezComponent_4 @ EditRezComponent.html:85
push../node_modules/@angular/core/fesm5/core.js.DebugContext_.logError @ core.js:30270
push../node_modules/@angular/core/fesm5/core.js.ErrorHandler.handleError @ core.js:4050
(anonymous) @ core.js:27136
ZoneDelegate.invoke @ zone.js:396
Zone.run @ zone.js:153
push../node_modules/@angular/core/fesm5/core.js.NgZone.runOutsideAngular @ core.js:26214
push../node_modules/@angular/core/fesm5/core.js.ApplicationRef.tick @ core.js:27136
(anonymous) @ core.js:26997
ZoneDelegate.invoke @ zone.js:396
onInvoke @ core.js:26255
ZoneDelegate.invoke @ zone.js:395
Zone.run @ zone.js:153
push../node_modules/@angular/core/fesm5/core.js.NgZone.run @ core.js:26169
next @ core.js:26997
schedulerFn @ core.js:23734
push../node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.__tryOrUnsub @ Subscriber.js:192
push../node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.next @ Subscriber.js:130
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber._next @ Subscriber.js:76
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:53
push../node_modules/rxjs/_esm5/internal/Subject.js.Subject.next @ Subject.js:47
push../node_modules/@angular/core/fesm5/core.js.EventEmitter.emit @ core.js:23718
checkStable @ core.js:26224
onHasTask @ core.js:26268
ZoneDelegate.hasTask @ zone.js:451
ZoneDelegate._updateTaskCount @ zone.js:472
Zone._updateTaskCount @ zone.js:294
Zone.runTask @ zone.js:215
drainMicroTaskQueue @ zone.js:611
ZoneTask.invokeTask @ zone.js:517
invokeTask @ zone.js:1671
globalZoneAwareCallback @ zone.js:1697
Show 2 more frames
EditRezComponent.html:85 ERROR CONTEXT DebugContext_ {view: {…}, nodeIndex: 8, nodeDef: {…}, elDef: {…}, elView: {…}}
```

**Comment by mrjohnr** [comment]
Inspired by [here ](https://stackoverflow.com/questions/49292932/expression-has-changed-error-on-opening-a-modal-popup-inside-a-component) seems I found a fix that works with 5.3.0,I prevent bsValueChange to fire while component loads
```
 (bsValueChange)="onDiChange(0,$event)"
...
  onDiChange(val: number, da: Date) {
        if (this.isLoading)
            return;
....
```
